src/main/java/org/intellij/sdk/codesync/files/ConfigFile.java

```diff
@@ -0,0 +1,137 @@
+
+        if (!configFile.isFile()) {
+            throw new InvalidConfigFileError("Config file path must be absolute path pointing to a file.");
+        }
+
+        this.configFile = configFile;
+        this.contentsMap = this.readYml();
+
+        try {
+            this.loadYmlContent();
+        } catch (InvalidConfigFileError e){
+            throw new InvalidConfigFileError("Config file is not valid.");
+        }
+    }
+
+    public Map<String, Object> getYMLAsHashMap() {
+        Map<String, Object> repos = new HashMap<>();
+        for (Map.Entry<String, ConfigRepo> repo : this.repos.entrySet()) {
+            repos.put(repo.getKey(), repo.getValue().getYMLAsHashMap());
+        }
+        Map<String, Object> configFile = new HashMap<>();
+        configFile.put("repos", repos);
+
+        return configFile;
+    }
+
+    private Map<String, Object> readYml() throws InvalidConfigFileError {
+        Yaml yaml = new Yaml();
+        InputStream inputStream;
+        try {
+            inputStream = new FileInputStream(this.configFile);
+        } catch (FileNotFoundException e) {
+            throw new InvalidConfigFileError("Config file must be present.");
+        }
+        String text = null;
+        try (Reader reader = new InputStreamReader(inputStream)) {
+            text = CharStreams.toString(reader);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+        return yaml.load(text);
+    }
+
+    private void writeYml() throws InvalidConfigFileError {
+        final DumperOptions options = new DumperOptions();
+        options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+        options.setPrettyFlow(true);
+
+        Yaml yaml = new Yaml(options);
+
+        // This line of code is placed here so that in  case of any error we do not write to the config file.
+        Map<String, Object> yamlConfig = this.getYMLAsHashMap();
+
+        FileWriter writer;
+        try {
+            writer = new FileWriter(this.configFile);
+        } catch (IOException e) {
+            throw new InvalidConfigFileError("Error while writing to the config file.");
+        }
+
+        yaml.dump(yamlConfig, writer);
+    }
+
+    public ConfigRepo getRepo(String repoPath) {
+        return this.repos.get(repoPath);
+    }
+
+    public void updateRepo(String repoPath, ConfigRepo newRepo) {
+        newRepo.lastSyncedAt = Utils.getCurrentDatetime();
+        this.repos.put(repoPath, newRepo);
+    }
+
+    public void publishRepoUpdate (ConfigRepo updatedRepo) throws InvalidConfigFileError {
+        this.reloadFromFile();
+        this.updateRepo(updatedRepo.repoPath, updatedRepo);
+        this.writeYml();
+    }
+
+    public void publishBranchUpdate (ConfigRepo updatedRepo, ConfigRepoBranch updatedBranch) throws InvalidConfigFileError {
+        this.reloadFromFile();
+        ConfigRepo configRepo = this.getRepo(updatedRepo.repoPath);
+        configRepo.lastSyncedAt = Utils.getCurrentDatetime();
```

Comment:
We are no longer using this field. We now have a separate file for last_sycned_at as it was only required by brew daemon